src/SFML/Network/Win32/SocketImpl.hpp

```diff
@@ -28,15 +28,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
-#ifdef _WIN32_WINDOWS
-    #undef _WIN32_WINDOWS
-#endif
-#ifdef _WIN32_WINNT
-    #undef _WIN32_WINNT
-#endif
-#define _WIN32_WINDOWS 0x0501
-#define _WIN32_WINNT   0x0501
 #include <SFML/Network/Socket.hpp>
+#include <SFML/Window/Win32/WindowsHeader.hpp>
```

Comment:
Wouldn't it make more sense to have the WindowsHeader.hpp file in the System module? Here the Network module depends on the Window module. It is just a header so it woks fine, but is seems odd to me.

Reply:
You're right, it's an invalid dependency (the Network module should not depend on the Window module), I thought it was the Window module only.

Reply:
OK, I will move it to `System/`.

Reply:
I have moved the header to `System/Win32`.

Reply:
@binary1248: your approval is still required for this PR

Reply:
You removed WindowsHeader.hpp instead of moving it in your last commit.